thirdparty/github.go

```diff
@@ -336,15 +340,33 @@ func getGithubClient(token, caller string, config retryConfig) *evergreen.GitHub
 // installation token is valid for a single GitHub API request.
 const defaultGitHubAPIRequestLifetime = 15 * time.Minute
 
-// getInstallationToken creates an installation token using Github app auth.
+// getCachedInstallationToken creates an installation token using Github app auth.
+// If possible, it will try to use an existing installation token for the app from the cache.
+// This should be used for all internal evergreen github operations.
 // If creating a token fails it will return the legacyToken.
-func getInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
+func getCachedInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
+	return installationToken(ctx, owner, repo, opts, true)
+}
+
+// GetInstallationToken creates an installation token using Github app auth.
+// It will not try to use the cache. This should only be used if there is a reason why a
+// cached token should not be used. If creating a token fails it will return the legacyToken.
+func GetInstallationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions) (string, error) {
+	return installationToken(ctx, owner, repo, opts, false)
+}
+
+func installationToken(ctx context.Context, owner, repo string, opts *github.InstallationTokenOptions, useCached bool) (string, error) {
```

Comment:
Re the "use a boolean" thing, I'd actually argue that this setup of using a boolean and making two wrapper functions is basically the exact same as calling one function vs. the other. The methods above already re-implement the `GetInstallationToken` and `GetCachedInstallationToken` methods, so I'm not sure they're really much better than the existing interface. I'd argue that unless we force all callers to use this method (and remove the other two `GetInstallationToken` and `getCachedInstallationToken` functions), this is just duplicating the existing methods from the top-level package.